### ISSUE TITLE
[Discovery] - Sentry Upgrade

### ISSUE THREAD

**Issue by alexpappasoddball** [department-of-veterans-affairs/va.gov-team#6075]
There is a new version of Sentry available.  This ticket is to look into if there are any considerations we need to make before upgrading.  If not, create a ticket for the actual work of upgrading.  Any considerations to upgrading Sentry should be noted here.

Also to consider here is revisiting the hosted sentry option and what it would take to do that, assuming we want to go that route.

**Comment by LindseySaari** [comment]
On-Premise Sentry 10 Upgrade

# Highlights
* [Recommended Way of Installing v 10](https://docs.sentry.io/server/installation/)
    * It looks like if we’re already running version 9.1, then the install script should take care of mostly everything. (We are on 9.1.1 according to devops repo)
* New visual capabilities
    * [New dashboards, a brand new Discover page, new search capabilities, a new events view, and cross-project issues.](https://blog.sentry.io/2019/03/06/new-features-greater-visibility)
    * Navigate across all issues in a single view (across all projects)
    * Graphs & Dashboards
    * Errors and events by release
* [Upgraded Integration System]( https://blog.sentry.io/2019/08/21/sentrys-internal-integrations-build-internal-tools-that-fit-your-workflow)
    *  Allows developers to build publicly available tools on top of Sentry
    *  Simpler authentication model for internal integrations. Instead of having to deal with OAuth at all, you receive an Access Token when you create an integration
    * [Ability to build our own integrations](https://blog.sentry.io/2019/04/23/introducing-sentry-integration-platform )
        * Done via Webhooks and the API or by the new UI elements feature
* New ability for rich UI elements    
    * [Link sentry issues to issues in your service (Slack, etc)](https://docs.sentry.io/workflow/integrations/integration-platform/)
* [Upgrade from 9 to 10](https://docs.sentry.io/server/upgrading/)
    *  If you’re upgrading to a new major release, its always recommended to start by generating a new configuration file (using the new version of Sentry). This will 	ensure that any new settings which may have been added are clearly visible and get configured correctly.
* What sentry 10 has to offer
    * https://github.com/getsentry/sentry/blob/master/CHANGES
    * https://blog.sentry.io/2020/01/07/self-hosted-sentry-10-is-ready-to-serve-get-it-while-its-hot
    *   https://github.com/getsentry/sentry/releases

# Breaking
* Migration issues may still exist. This relevant issue affected users upgrading from 9.1.2) https://github.com/getsentry/onpremise/issues/366
* [List of all open issues currently]( https://github.com/getsentry/sentry/issues?page=1&q=is%3Aissue+is%3Aopen)
* [Current slack issues](https://github.com/getsentry/sentry/issues?utf8=%E2%9C%93&q=is%3Aissue+is%3Aopen+Slack)
    * TL;DR: Issues with alerting rules & slack channels


# Of Concern
* [Snuba](https://blog.sentry.io/2019/05/16/introducing-snuba-sentrys-new-search-infrastructure) - which is a new storage and query service for event data is new to Sentry 10. Snuba has a lot of dependencies (Clickhouse and Kafka, which come with some new dependencies like Zookeeper, which comes with other). This makes sentry more capable, but we may run into dependency issues. 
* [Another Snuba Issue](https://forum.sentry.io/t/migrate-sentry-9-issues-events-to-sentry-10-w-snuba/6893) - pg_dump of v9 database restores everything except for issues and events. No current way  to migrate my v9 db events and issues into v10’s Snuba system
* Current [Forum](https://forum.sentry.io/)


# Other Links
* [Changes between 9.1.x -> 10.0](https://github.com/getsentry/sentry/compare/10.0.0...releases/9.1.x)
* [Ticket from last upgrade](https://app.zenhub.com/workspaces/vft-59c95ae5fda7577a9b3184f8/issues/department-of-veterans-affairs/vets.gov-team/15119)


# Local install issues
* Running into an issue when running "https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/sentry/README.md#commands" from the link [here](https://github.com/department-of-veterans-affairs/devops/blob/master/ansible/deployment/config/sentry/README.md#commands)
